src/main/java/com/example/Storehouse/builder/multiproduct/MultiProductResponseBuilder.java

```diff
@@ -0,0 +1,81 @@
+package com.example.Storehouse.builder.multiproduct;
+
+import com.example.Storehouse.builder.ResponseBuilder;
+import com.example.Storehouse.exception.exceptions.NullOptionalException;
+import com.example.Storehouse.exception.exceptions.ProductNotFoundException;
+import com.example.Storehouse.exception.exceptions.WrongQuantityException;
+import com.example.Storehouse.models.product.Product;
+import com.example.Storehouse.models.product.ProductRequest;
+import com.example.Storehouse.models.request.RequestMultiProducts;
+import com.example.Storehouse.models.response.multiProductsResponse.MultiProductsResponse;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component
+public class MultiProductResponseBuilder extends ResponseBuilder implements iMultiProductResponseBuilder {
+
+    public MultiProductResponseBuilder() {
+    }
+
+    public void dataProcessing(RequestMultiProducts request) {
+        multiProductsResponse.get().getProgetListOfProducts().clear();
+        this.requestData = Optional.of(request);
+        multiProductsResponse.get().setId(requestData.get().getId());
+        print.info("New request. " + request);
+    }
+
+
+    public void foundProductInDataBase(String name) {
```

Comment:
Remove these method to proper RepositoryService, if you dont have then create,remember what we spoke about Single responsobility here you have mized logic with database work, and building models. separate them

Reply:
done